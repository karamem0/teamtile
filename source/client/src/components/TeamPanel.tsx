import React from 'react';
import AppContext from '../contexts/AppContext';
import useTeams from '../hooks/useTeams';
import LoaderPanel from './LoaderPanel';
import EmptyPanel from './EmptyPanel';
import TeamCard from './TeamCard';

const TeamPanel: React.FC = () => {

  const [ token, , , setError ] = React.useContext(AppContext);
  const [ teams, error ] = useTeams({ token });

  React.useEffect(() => {
    if (!setError) {
      return;
    }
    setError(error);
  }, [ setError, error ]);

  if (!teams) {
    return (
      <LoaderPanel />
    );
  }

  if (!teams.length) {
    return (
      <EmptyPanel />
    );
  }

  return (
    <div className="panel panel-grid">
      {
        teams.map((team) =>
          <TeamCard
            key={team.id}
            team={team} />
        )
      }
    </div>
  );

};

export default TeamPanel;
