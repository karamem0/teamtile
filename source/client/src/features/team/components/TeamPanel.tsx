//
// Copyright (c) 2021-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { setFilter, setLoading } from '../../../stores/Action';
import { Event } from '../../../types/Event';
import { clearCache } from '../managers/TeamManager';
import messages from '../messages';
import { useDebounce } from 'react-use';
import { useIntl } from 'react-intl';
import { useStore } from '../../../providers/StoreProvider';

import Presenter from './TeamPanel.presenter';

function TeamPanel() {

  const { dispatch } = useStore();
  const intl = useIntl();
  const [ cache, setCache ] = React.useState<string>();

  useDebounce(() => {
    dispatch(setFilter(cache));
  }, 500, [
    cache,
    dispatch
  ]);

  const handleFilterChange = React.useCallback((_: Event, data?: string) => {
    setCache(data);
  }, []);

  const handleLinkToGitHub = React.useCallback(() => {
    window.open(intl.formatMessage(messages.GitHubLink), '_blank', 'noreferrer');
  }, [
    intl
  ]);

  const handleLinkToPrivacyPolicy = React.useCallback(() => {
    window.open(intl.formatMessage(messages.PrivacyPolicyLink), '_blank', 'noreferrer');
  }, [
    intl
  ]);

  const handleLinkToTermsOfUse = React.useCallback(() => {
    window.open(intl.formatMessage(messages.TermsOfUseLink), '_blank', 'noreferrer');
  }, [
    intl
  ]);

  const handleRefreshClick = React.useCallback(async (event: Event) => {
    const { shiftKey } = event as KeyboardEvent;
    if (shiftKey) {
      await clearCache();
    }
    dispatch(setLoading(true));
  }, [
    dispatch
  ]);

  return (
    <Presenter
      onFilterChange={handleFilterChange}
      onLinkToGitHub={handleLinkToGitHub}
      onLinkToPrivacyPolicy={handleLinkToPrivacyPolicy}
      onLinkToTermsOfUse={handleLinkToTermsOfUse}
      onRefreshClick={handleRefreshClick} />
  );

}

export default TeamPanel;
