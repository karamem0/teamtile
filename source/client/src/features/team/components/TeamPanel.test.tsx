//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render } from '@testing-library/react';

import TeamPanel from './TeamPanel.presenter';

jest.mock('./TeamGrid', () =>
  function TeamGrid({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="TeamGrid">
        {children}
      </div>
    );
  });

test('create shapshot', async () => {
  const params = {};
  const { asFragment } = render(<TeamPanel {...params} />);
  expect(asFragment()).toMatchSnapshot();
});
