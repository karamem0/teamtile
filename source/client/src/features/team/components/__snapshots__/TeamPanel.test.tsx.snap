// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: column;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  grid-gap: 0.5rem;
}

.emotion-1 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: 1fr auto;
  grid-gap: 0.25rem;
}

.emotion-2 {
  min-height: 2.25rem;
}

@media (width >= 600px) {
  .emotion-2 {
    max-width: 20rem;
  }
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span
        class="fui-Input r1oeeo9n fui-SearchBox emotion-2 ___13yg2oi_15g8mz3 f1f5gg8d fhxju0i fvcxoqz f1ub3y4t f1l4zc64 f1m52nbi f8vnjqi fz1etlk f1klwx88 f1hc16gm fjuset5 f1xzfw5u"
      >
        <span
          class="fui-Input__contentBefore r1572tok fui-SearchBox__contentBefore"
        >
          <svg
            aria-hidden="true"
            class="___12fm75w_1fmvgpr f1w7gpdv fez10in fg4l7m0"
            fill="currentColor"
            height="1em"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.73 13.44a6.5 6.5 0 1 1 .7-.7l3.42 3.4a.5.5 0 0 1-.63.77l-.07-.06-3.42-3.41Zm-.71-.71A5.54 5.54 0 0 0 14 8.5a5.5 5.5 0 1 0-1.98 4.23Z"
              fill="currentColor"
            />
          </svg>
        </span>
        <input
          class="fui-Input__input r12stul0 fui-SearchBox__input ___69a85s0_ca35df0 fk8j09s f1vdfbxk f18izjht fcoa6sg"
          placeholder="Search teams"
          type="search"
          value=""
        />
        <span
          class="fui-Input__contentAfter r1572tok fui-SearchBox__contentAfter ___yrbim60_1xrsupy f1cnd47f f1ufnopg fniina8 f3tsq5r f1a3p1vp"
        >
          <span
            aria-label="clear"
            class="fui-SearchBox__dismiss r1pvzcuu"
            role="button"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_1fmvgpr f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m4.09 4.22.06-.07a.5.5 0 0 1 .63-.06l.07.06L10 9.29l5.15-5.14a.5.5 0 0 1 .63-.06l.07.06c.18.17.2.44.06.63l-.06.07L10.71 10l5.14 5.15c.18.17.2.44.06.63l-.06.07a.5.5 0 0 1-.63.06l-.07-.06L10 10.71l-5.15 5.14a.5.5 0 0 1-.63.06l-.07-.06a.5.5 0 0 1-.06-.63l.06-.07L9.29 10 4.15 4.85a.5.5 0 0 1-.06-.63l.06-.07-.06.07Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </span>
      </span>
      <button
        aria-haspopup="menu"
        class="fui-Button r1alrhcs ___cpe5h10_1yhfyj7 f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f ft1hn21 fuxngvv fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1dfjoow f1j98vj9 f1tme0vf f4xjyn1 f18onu3q f9ddjv3 f18ktai2 fwbmr0d f44c6la"
        id="menur0"
        type="button"
      >
        <span
          class="fui-Button__icon rywnvv2"
        >
          <svg
            aria-hidden="true"
            class="___12fm75w_1fmvgpr f1w7gpdv fez10in fg4l7m0"
            fill="currentColor"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 5a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm-1 3a1 1 0 1 0 2 0 1 1 0 0 0-2 0Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      data-testid="test-TeamGrid"
    />
    <div
      data-testid="test-DrawerFactory"
    />
  </div>
</DocumentFragment>
`;
